proposals/route-blinding.md

```diff
@@ -0,0 +1,285 @@
+sender. For route blinding it only needs to contain the `scid` to use when forwarding the message,
+but it may be extended with additional data in the future. It uses ChaCha20-Poly1305 as AEAD scheme.
+
+1. type: 10 (`encrypted_recipient_data`)
+2. data:
+    * [`...*byte`:`encrypted_recipient_data`]
+
+Once decrypted, the content of this encrypted payload is a TLV stream that may contain any tlv
+record defined in Bolt 4 (onion TLV namespace).
+
+### Creating a blinded route
+
+`N(r)` performs the following steps to create a blinded route:
+
+```text
+Initialization:
+
+  e(0) <- {0;1}^256
+  E(0) = e(0) * G
+
+Blinding:
+
+  For i = 0 to r-1:
+    ss(i) = H(e(i) * P(i)) = H(k(i) * E(i))         // shared secret known only by N(r) and N(i)
+    B(i) = HMAC256("blinded_node_id", ss(i)) * P(i) // Blinded node_id for N(i), private key known only by N(i)
+    rho(i) = HMAC256("rho", ss(i))                  // Key used to encrypt payload for N(i) by N(r)
+    e(i+1) = H(E(i) || ss(i)) * e(i)                // Ephemeral private key, only known by N(r)
+    E(i+1) = H(E(i) || ss(i)) * E(i)                // NB: N(i) must not learn e(i)
+
+Blinded route:
+  
+  (P(0),fees(0),cltv(0),encrypted_recipient_data(0))
+  (B(1),fees(1),cltv(1),encrypted_recipient_data(1))
+  ...
+  (B(r-1),fees(r-1),cltv(r-1),encrypted_recipient_data(r-1))
+```
+
+Note that this is exactly the same construction as Sphinx, but at each hop we use the shared secret
+to derive a blinded `node_id` for `N(i)` for which the private key will only be known by `N(i)`.
+
+The recipient needs to provide `E(0)` and the blinded route to potential senders.
+The `encrypted_recipient_data(i)` is encrypted with ChaCha20-Poly1305 using the `rho(i)` key, and
+contains the real `short_channel_id` to forward to (and potentially other fields).
+`E(i)` is included as additional authenticated data to detect probing attempts by the sender.
+
+Note that the introduction point uses the real `node_id`, not the blinded one, because the sender
+needs to be able to locate this introduction point and find a route to it. But the sender will send
+`E(0)`, which will allow the introduction point to compute the shared secret and correctly forward.
+
+Note that in the specific case of payments, the recipient can sign the invoice with `e(0)`.
+The sender will recover `E(0)` from the signature so no extra field needs to be added to Bolt 11.
+And this ensures the recipient doesn't reveal his real `node_id` through the invoice signature.
+
+However, if the recipient wants to be able to prove invoice ownership in the future, she should
+sign the invoice with a different key and provide `E(0)` via a dedicated Bolt 11 field.
+
+### Sending to a blinded route
+
+The sender finds a route to the introduction point `N(0)`, and extends it with the blinded route.
+It then creates an onion for that route, and includes `E(0)` and `encrypted_recipient_data(0)` in
+the onion payload for `N(0)`.
```

Comment:
I don't think this attack works, because for the decryption to work at Carol's node with `encrd(Y)`, X's private key is needed to generate the right `E(Y)`. Otherwise Carol will simply fail to decrypt which doesn't tell the attacker anything (AFAICT).

Specifically your step 4 will result in Carol sending back an `update_fail_malformed_htlc`, not a `temporary_channel_failure`.